sdk/identity/Azure.Identity/src/Azure.Identity.csproj

```diff
@@ -10,6 +10,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" />
```

Comment:
This is not an approved dependency for our Track 2 (Azure.*) libraries. Would it be possible to output the expiration in addition to the access token so we don't have to parse the JWT?

Reply:
Yes, this was challenging for me. 

I didn't want to add a new dependency in this specific library, but I saw it is used by the other libraries in this solution like ServiceBus, ContainerRegistry. 

The problem is that there's not a GetToken cmdlet (like with the Azure CLI with _az account get-access-token_), and the method I call from PowerShell only returns the token, so I have to parse the expiration time from it.

The real solution should be to ask the Azure PowerShell team to (finally) add a GetToken cmdlet feature, this will make the implementation here more easier with less logic in the PowerShell script part.

If you have a better way to parse the expiration from the token without adding the dependency, please let me know, I will implement it.

Reply:
@schaabs good news! 
I was looking in the Azure PowerShell code again and I see that they released a new version (2.2.0)  with a Get-AzAccessToken cmdlet, where they return also the expiration 👍 💯 . 

Now the question...how to deal with this? 

I can check in the powershell script running if the module loaded is equal or greater than 2.2.0. and use the new Get-AzAccessToken cmdlet.

But..If not, should I still use the current approach to get the token? This will mean to still have a dependecy to the System.IdentityModel.Tokens.Jwt package to parse the token to get the expiration.

Or we can decide to depend only with the 5.1.0 Az Accounts Module and forward?

So the choices are:

1. Depend from Az.Accounts 2.2.0 and forward only (Throwing a message like "Az Accounts >=2.2.0 is required", if not found) 
2. Implement logic the deal with Az.Accounts < 2.2.0 and >=2.2.0:
   - if < 2.2.0 use the current implementation of this PR (and depending on the System.IdentityModel.Tokens.Jwt package)
   - if >= 2.2.0 use the Get-AzAccessToken cmdlet